While running the test_index_timeout python test in the sdc-2i-timeout branch, I'm getting something unexpected. Is this what you were seeing? The client is not raising an exception with timeout=1. 

# 

## FAIL: test_index_timeout (riak.tests.test_all.RiakHttpTransportTestCase)

Traceback (most recent call last):
  File "riak/tests/test_2i.py", line 422, in test_index_timeout
    bucket.get_index('field1_bin', 'val1', timeout=1)
AssertionError: RiakError not raised

--- a/src/riak_kv_wm_index.erl
+++ b/src/riak_kv_wm_index.erl
@@ -258,12 +258,10 @@ whack_index_fsm(ReqID, Pid) ->
     clear_index_fsm_msgs(ReqID).
 
 wait_for_death(Pid) ->
+    Ref = erlang:monitor(process, Pid),
     exit(Pid, kill),
-    case is_process_alive(Pid) of
-        true ->
-            timer:sleep(10),
-            wait_for_death(Pid);
-        false ->
+    receive
+        {'DOWN', Ref, process, Pid, _Info} ->
             ok
     end.
 

